server/data/data-api/src/main/java/org/apache/james/sieverepository/api/SieveQuotaRepository.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *   Licensed to the Apache Software Foundation (ASF) under one
+ *   or more contributor license agreements.  See the NOTICE file
+ *   distributed with this work for additional information
+ *   regarding copyright ownership.  The ASF licenses this file
+ *   to you under the Apache License, Version 2.0 (the
+ *   "License"); you may not use this file except in compliance
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ *
+ */
+
+package org.apache.james.sieverepository.api;
+
+import org.apache.james.sieverepository.api.exception.QuotaNotFoundException;
+import org.apache.james.sieverepository.api.exception.StorageException;
+
+/**
+ * SieveQuotaRepository
+ */
+public interface SieveQuotaRepository {
+
+    boolean hasQuota() throws StorageException;
+
+    long getQuota() throws QuotaNotFoundException, StorageException;
+
+    void setQuota(long quota) throws StorageException;
```

Comment:
I guess it's a quota about the storage size, there QuotaSize type for that

Reply:
Very true, but it's in mailbox project. Should we generify this?

Also, once we are at it, maybe strong typing the entire SieveRepository might be a good move (I did not try to do it here).